repository/src/main/java/org/apache/atlas/repository/graph/GraphHelper.java

```diff
@@ -376,6 +376,30 @@ public static AtlasVertex getClassificationVertex(AtlasVertex entityVertex, Stri
         return ret;
     }
 
+    public static List<AtlasVertex> getAllClassificationVertex(AtlasGraph graph, String classificationName) {
+        Iterable vertices = graph.query().has(TYPE_NAME_PROPERTY_KEY, classificationName).vertices();
+        if (vertices == null) {
+            return Collections.emptyList();
+        }
+        return IteratorUtils.toList(vertices.iterator());
+    }
+
+    public static List<AtlasVertex> getAllAssetsWithClassificationAttached(AtlasGraph graph, String classificationName, int limit) {
```

Comment:
Can there be a case where the value if PROPAGATED_TRAIT_NAMES_PROPERTY_KEY or TRAIT_NAMES_PROPERTY_KEY be incorrect? If yes, we should leverage edges to get the asset vertices.

Reply:
It shouldn't except we have some bug in there, we have instances of assets have this fields when classification was not attached not the other way around(Although this case would me more visible in product UI so people reported more)

Now this will help on those cases also where there PROPAGATED_TRAIT_NAMES_PROPERTY_KEY was wrongly there, it will fix those cases automatically.